### PR TITLE
[Java] Don't load cpp library in dev model.

### PR DESCRIPTION
**Pull request by XiaodongLv**
## Why are these changes needed?
Don't load cpp library in dev model, because it will be error when nativeGetSystemConfig is invoked in local model on the Mac.

## Checks

- [x] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: java/runtime/src/main/java/io/ray/runtime/util/SystemConfig.java [hunk 1]
```diff
@@ -2,17 +2,38 @@
...
 
+  private static RayConfig rayConfig = null;
+
   private static Gson gson = new Gson();
 
   /// A cache to avoid duplicated reading via JNI.
   private static HashMap<String, Object> cachedConfigs = new HashMap<>();
 
+  /**
+   * The string key to use for getting the large size passed by value. If the the size of an
```
Comment (jovany-wang) [inline java/runtime/src/main/java/io/ray/runtime/util/SystemConfig.java:20]:
```suggestion
   * The string key to use for getting the largest size passed by value. If the the size of an
```

File: java/test/src/main/java/io/ray/test/UserLoggerTest.java [hunk 2]
```diff
@@ -28,13 +27,13 @@ public class UserLoggerTest extends BaseTest {
   public void setupJobConfig() {
     System.setProperty("ray.job.jvm-options.0", "-Dray.logging.loggers.0.name=test_user_logger1");
     System.setProperty(
-        "ray.job.jvm-options.1", "-Dray.logging.loggers.0.file-name=test_user_logger-1-%p");
+        "ray.job.jvm-options.1", "-Dray.logging.loggers.0.file-name=test_user_logger-1-%p-%j");
```
Comment (jovany-wang) [inline java/test/src/main/java/io/ray/test/UserLoggerTest.java:30]: I believe these are unintended changes.

**Review by jovany-wang — CHANGES_REQUESTED** [review]
Please revert the unnecessary changes.

File: java/test/src/main/java/io/ray/test/UserLoggerTest.java (hunk 2 above)
Comment (XiaodongLv) [reply java/test/src/main/java/io/ray/test/UserLoggerTest.java]: Sorry, I cut branch from last pr rather than master

**Review by jovany-wang — APPROVED** [review]

**Pull request merged**